pkg/uploadserver/uploadserver.go

```diff
@@ -400,14 +407,15 @@ func newAsyncUploadStreamProcessor(stream io.ReadCloser, dest, imageSize string,
 	return processor, processor.ProcessDataWithPause()
 }
 
-func newUploadStreamProcessor(stream io.ReadCloser, dest, imageSize string, filesystemOverhead float64, preallocation bool, contentType string) error {
+func newUploadStreamProcessor(stream io.ReadCloser, dest, imageSize string, filesystemOverhead float64, preallocation bool, contentType string) (string, error) {
 	if contentType == common.FilesystemCloneContentType {
-		return filesystemCloneProcessor(stream, common.ImporterVolumePath)
+		return "false", filesystemCloneProcessor(stream, common.ImporterVolumePath)
```

Comment:
Is there a strong reason to use strings that are booleans? it's a bummer to lose the benefit of type-checking

Reply:
It's "true", "false" or "skipped".

I could do a type redefinition here (like `type PreallocationStatus string`), but it seemed to me like an overkill.

Reply:
In Go, in general it is very simple to do the redefinition, and it adds some type safety, in particular with typos in the strings. maybe you can do:
```go
type Preallocation string

const PreallocationYes Preallocation="true"
const PreallocationNo Preallocation="false"
const PreallocationSkip Preallocation="skipped"
```
And you can pass the Yes/No instead of the booleans, and return Yes/No/Skip for the applied. I think that would make it easier to read. I had to re-read a few times before I found the place where the value was set to none.

Reply:
This is a cherry-pick for 1.28. I don't want to change this here, let me create a new PR for that.